### DISCUSSION TITLE
Correctly updating software - best practice

### DISCUSSION THREAD

**Discussion by ghost** [retronas/discussions#84] [Q&A]
Hi,
I wonder how to, according to best practice, update retronas. 
I have a raspberry pi 4 and I have managed to locate 3 places that I need to be aware of if I want to update the software.

1. Cockpit GUI - Security updates?
2. Raspberry pi software
3. Retronas 

What is the correct order?

Thank you

**Comment by sairuk** [comment]
This has been converted to a discussion of type Q&A

To answer your question there is no preferred order, but i will give you some general guidelines

- **RetroNAS** will self update every time the text user interface is run and/or you can do so manually from the Update menu 
- **Cockpit GUI**, in most cases they should be updated as soon as they are available, these are Linux OS package updates which will include security fixes, best practice here would be to review and understand each package update but I expect much like Windows Updates people will blindly click Install All.
- **Raspberry PI** software insofar as the firmware(?) through `rpi-update`? I believe the RPI team still advise only updating with good reason (e.g. something is broken for you), someone can correct me if thatis not the case

The only time I have updated RPI firmware purposely was while I was using my RPI4 as a desktop and needed later firmware to fix my very particular dual-screen+audio setup, which they subsequently broke in a number of other updates and I had to roll back while continuously testing attempted bug fixes.

As per the `rpi-update` github readme

_"Bug fixes and improvements will eventually make their way into new Raspberry Pi OS releases and apt-get when they are considered sufficiently well tested."_

So when stable the Cockpit-UI updates will pickup the applicable RPI updates.

We should probably provide some doco on this, thanks for asking the question

**Reply by sairuk** [reply]
We'll also need to provide a means for people to trigger system updates from the text ui, we don't have that

**Reply by sairuk** [reply]
I've added GNU/Linux OS update options to the updates menu for the text-ui
